strongbox-web-core/src/main/java/org/carlspring/strongbox/app/boot/BootProgressServlet.java

```diff
@@ -0,0 +1,98 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import io.reactivex.disposables.Disposable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+
+public class BootProgressServlet
+        extends HttpServlet
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(BootProgressServlet.class);
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException
+    {
+        try
+        {
+            if (!MediaType.TEXT_EVENT_STREAM_VALUE.equalsIgnoreCase(request.getHeader("Accept")))
+            {
+                response.sendError(HttpStatus.NOT_ACCEPTABLE.value());
+                return;
+            }
+
+            response.setStatus(HttpStatus.OK.value());
+            response.setContentType(MediaType.TEXT_EVENT_STREAM_VALUE);
+            response.setCharacterEncoding("UTF-8");
+
+            response.setHeader("Connection", "keep-alive");
+            response.setHeader("Cache-control", "no-cache, no-store, max-age=0, must-revalidate");
+            response.setHeader("Pragma", "no-cache");
+            response.setHeader("Expires", "0");
+            // -- CORS - allow everybody while booting, but set max-age to force browsers re-check after boot.
+            response.setHeader("Allow", "GET, OPTIONS");
+            response.setHeader("Access-Control-Allow-Origin", "*");
+            response.setHeader("Access-Control-Allow-Headers", "*");
+            response.setHeader("Access-Control-Max-Age", "600");
+
+            final AsyncContext asyncContext = request.startAsync(request, response);
+
+            final Disposable disposable = BootProgressBeanPostProcessor.getProgressObservable()
+                                                                       .doOnComplete(() -> {
+                                                                           write("ready", "", asyncContext);
+                                                                           asyncContext.getResponse().getWriter().flush();
+                                                                           asyncContext.complete();
+
+                                                                       })
+                                                                       .doOnError((e) -> {
+                                                                           // IOException occurs mostly when the client
+                                                                           // closes the connection unexpectedly on
+                                                                           // their end.
+                                                                           if (!(e instanceof IOException))
```

Comment:
`if (!(e instanceof IOException))` I'm not sure if we should skip all `IOException`s

Reply:
Okay, removed. :)